packages/flutter/test/material/bottom_app_bar_test.dart

```diff
@@ -38,6 +38,49 @@ void main() {
     );
   });
 
+  testWidgets('custom shape', (WidgetTester tester) async {
+    final Key key = UniqueKey();
+    Future<void> pump(FloatingActionButtonLocation location) async {
+      await tester.pumpWidget(
+        SizedBox(
+          width: 200,
+          height: 200,
+          child: RepaintBoundary(
+            key: key,
+            child: MaterialApp(
+              home: Scaffold(
+                floatingActionButton: FloatingActionButton(
+                  onPressed: () { },
+                ),
+                floatingActionButtonLocation: location,
+                bottomNavigationBar: BottomAppBar(
+                  shape: AutomaticNotchedShape(
```

Comment:
I have been searching for an hour on how to add custom shape finally I got it thanks